cylc/flow/id_match.py

```diff
@@ -234,6 +240,102 @@ def filter_ids(
     return ret, _not_matched
 
 
+def filter_ids_2(
+    xpool: 'Set[Tuple[str, PointBase]]',
```

Comment:
Two use cases:

* Matching tasks in a pool.
* Matching tasks in a hold list.

Both use cases are essentially looping over a list and looking for matching entries.

If we reformatted the hold-list to match the format of the pool, could we use the one method for both cases to avoid adding a new implementation?

Reply:
(Yes, I meant to come back to this myself, will take another look...)

Reply:
Done.